File: pocketr-java.api/src/main/java/com/decrux/pocketr/api/entities/dtos/AccountBalanceTimeseriesDto.java
package com.decrux.pocketr.api.entities.dtos;

import java.util.List;
import java.util.UUID;

public record AccountBalanceTimeseriesDto(
    UUID accountId,
    String accountName,
    String accountType,
    String currency,
    List<BalanceTimeseriesPointDto> points
) {
    public UUID getAccountId() {
        return accountId;
    }

    public String getAccountName() {
        return accountName;
    }

    public String getAccountType() {
        return accountType;
    }

    public String getCurrency() {
        return currency;
    }

    public List<BalanceTimeseriesPointDto> getPoints() {
        return points;
    }
}

File: pocketr-api/src/main/kotlin/com/decrux/pocketr/api/entities/dtos/AccountBalanceTimeseriesDto.kt
package com.decrux.pocketr.api.entities.dtos

import java.util.UUID

data class AccountBalanceTimeseriesDto(
    val accountId: UUID,
    val accountName: String,
    val accountType: String,
    val currency: String,
    val points: List<BalanceTimeseriesPointDto>,
)
